test/org.jkiss.dbeaver.test.platform/src/org/jkiss/dbeaver/model/SQLUtilsTest.java

```diff
@@ -68,39 +69,130 @@ public void extractProcedureParameterTypes_whenNullOrEmpty_thenParens() {
+        Assert.assertEquals(
+            "(NUMBER)",
+            SQLUtils.extractProcedureParameterTypes("id NUMBER")
+        );
+        Assert.assertEquals(
+            "(ARRAY(VARCHAR))",
+            SQLUtils.extractProcedureParameterTypes("arr ARRAY(VARCHAR)")
+        );
     }
 
     @Test
     public void extractProcedureParameterTypes_whenWhitespaceAndCase_thenCanonicalUpper() {
-        Assert.assertEquals("(NUMBER, VARCHAR)",
-            SQLUtils.extractProcedureParameterTypes("(  a   number  ,   b   varchar  )"));
-        Assert.assertEquals("(ARRAY, OBJECT, VARIANT)",
-            SQLUtils.extractProcedureParameterTypes("( arr  array , obj   object , v   variant )"));
+        Assert.assertEquals(
+            "(NUMBER, VARCHAR)",
+            SQLUtils.extractProcedureParameterTypes("(  a   number  ,   b   varchar  )")
+        );
+        Assert.assertEquals(
+            "(ARRAY, OBJECT, VARIANT)",
+            SQLUtils.extractProcedureParameterTypes("( arr  array , obj   object , v   variant )")
+        );
     }
 
     @Test
     public void extractProcedureParameterTypes_whenComplexSignature_thenCorrectSplit() {
-        Assert.assertEquals("(DECIMAL(38,0), ARRAY(DECIMAL(10,2)), OBJECT)",
-            SQLUtils.extractProcedureParameterTypes("(a DECIMAL(38,0), b ARRAY(DECIMAL(10,2)), c OBJECT)"));
+        Assert.assertEquals(
+            "(DECIMAL(38,0), ARRAY(DECIMAL(10,2)), OBJECT)",
+            SQLUtils.extractProcedureParameterTypes("(a DECIMAL(38,0), b ARRAY(DECIMAL(10,2)), c OBJECT)")
+        );
+    }
+
+    @Test
+    public void addMultiStatementDDL_whenNullOrEmpty_thenNoChange() {
+        StringBuilder sb = new StringBuilder();
+        SQLUtils.addMultiStatementDDL(new String[] {";"}, sb, null);
+        Assert.assertEquals("", sb.toString());
+
+        SQLUtils.addMultiStatementDDL(new String[] {";"}, sb, "   ");
+        Assert.assertEquals("", sb.toString());
+    }
+
+    @Test
+    public void addMultiStatementDDL_oracleUserAndGrants_examples() {
+        String ddl = "CREATE USER \"TEST_USER_DECL\" IDENTIFIED BY VALUES 'S:91A176BDA85169E00F18E6F4C9FDE5609E82'\n" +
+            "      DEFAULT TABLESPACE \"USERS\"\n" +
+            "      TEMPORARY TABLESPACE \"TEMP\";\n" +
+            "GRANT CREATE TABLE TO \"TEST_USER_DECL\";\n" +
+            "GRANT CREATE SESSION TO \"TEST_USER_DECL\";\n\n" +
+            "GRANT \"ROLE1\" TO \"TEST_USER_DECL\";\n" +
+            "GRANT \"ROLE2\" TO \"TEST_USER_DECL\";\n" +
+            "GRANT \"ROLE3\" TO \"TEST_USER_DECL\";\n";
+
+        StringBuilder sb = new StringBuilder();
+        SQLUtils.addMultiStatementDDL(new String[] {";"}, sb, ddl);
+
+        Assert.assertEquals(
+            "CREATE USER \"TEST_USER_DECL\" IDENTIFIED BY VALUES 'S:91A176BDA85169E00F18E6F4C9FDE5609E82';\n" +
+                "DEFAULT TABLESPACE \"USERS\";\n" +
+                "TEMPORARY TABLESPACE \"TEMP\";\n" +
```

Comment:
This test is not valid. It must be a single, contiguous statement, not three separate ones. See https://docs.oracle.com/cd/B13789_01/server.101/b10759/statements_8003.htm